src/js/providers/providers.js

```diff
@@ -6,33 +6,51 @@ define([
     ], function(html5, flash, youtube, _) {
 
 
-    function chooseProvider(source) {
-        // prevent throw on missing source
-        source = _.isObject(source) ? source : {};
+    function Providers(primary) {
+        this.providers = Providers.defaultList.slice();
 
-        var chosen = _.find(providers, function (provider) {
-            return provider.supports(source);
-        });
-
-        return chosen;
+        if (primary === 'flash') {
+            swap(this.providers, html5, flash);
+        }
     }
 
-    var priority = function(p) {
-        var idx = _.indexOf(providers, p);
-        if (idx < 0) {
-            // No provider matched
-            return Number.MIN_VALUE;
+    // When choosing a provider, go through this array
+    //   and select the first that works for the source
+    Providers.defaultList = [html5, flash, youtube];
+
+    _.extend(Providers.prototype, {
```

Comment:
Not sure what the experts say, but Function.prototype = {...} works, and on the surface is fewer operations.